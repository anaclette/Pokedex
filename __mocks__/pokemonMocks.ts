import {
  Species,
  Ability,
  Move,
  Sprites,
  Stat,
  Type,
  PokemonFullDetails,
  NewListPokemon,
} from '../src/types/Pokemon';

const speciesMock: Species = {
  name: 'Pikachu',
  url: 'url',
};

const abilitiesMock: Ability = {
  ability: speciesMock,
  is_hidden: false,
  slot: 1,
};

const movesMock: Move = {
  move: speciesMock,
};

const spritesMock: Sprites = {
  back_default: 'back-default-url',
  back_female: null,
  back_shiny: 'back-shiny-url',
  back_shiny_female: null,
  front_default: 'front-default-url',
  front_female: null,
  front_shiny: 'front-shiny-url',
  front_shiny_female: null,
};

const statsMock: Stat = {
  base_stat: 20,
  effort: 30,
  stat: speciesMock,
};

const typesMock: Type = {
  slot: 1,
  type: speciesMock,
};

const pokemonMock: PokemonFullDetails = {
  abilities: [abilitiesMock],
  base_experience: 50,
  forms: [speciesMock],
  height: 50,
  held_items: [],
  id: '3',
  is_default: false,
  location_area_encounters: 'location-area-encounters',
  moves: [movesMock],
  name: 'Pikachu',
  order: 10,
  past_types: [],
  species: speciesMock,
  sprites: spritesMock,
  stats: [statsMock],
  types: [typesMock],
  weight: 100,
};

const mockNewListPokemon: NewListPokemon = {
  id: '20',
  name: 'Metapod',
  picture: 'url',
  imgColor: 'peru',
  textColor: 'lime',
};

export {pokemonMock, abilitiesMock, speciesMock, mockNewListPokemon};
