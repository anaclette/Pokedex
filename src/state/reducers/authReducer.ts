import {createSlice, PayloadAction} from '@reduxjs/toolkit';
import {STATE_MODULES} from '../../common/constants';

type Auth = {username?: string | null | undefined};

const initialState: Auth = {
  username: null,
};

const authReducer = createSlice({
  name: STATE_MODULES.AUTH,
  initialState,
  reducers: {
    logIn: (state, action: PayloadAction<string>) => ({
      ...state,
      username: action.payload,
    }),
    logOut: () => initialState,
  },
});

export const {logIn, logOut} = authReducer.actions;

export default authReducer.reducer;
