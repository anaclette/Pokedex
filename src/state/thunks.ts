import {createAsyncThunk} from '@reduxjs/toolkit';
import {ACTIONS, baseUrl, STATE_MODULES} from '../common/constants';

export const getPokemons = createAsyncThunk(
  `${STATE_MODULES.POKEMONS}/${ACTIONS.GET_POKEMONS}`,
  async (_, {rejectWithValue}) => {
    try {
      const pokemons = await fetch(
        `${baseUrl}/${STATE_MODULES.POKEMONS}?limit=50&offset=50`,
      )
        .then(response => response.json())
        .then(data => data);
      return pokemons;
    } catch (error) {
      rejectWithValue(error);
    }
  },
);
