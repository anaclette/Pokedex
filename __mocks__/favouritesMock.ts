import {FavouritesSlice} from '../src/state/reducers/favouritesReducer';
import {mockNewListPokemon} from './pokemonMocks';

const Favourites: FavouritesSlice = {
  data: [mockNewListPokemon],
  favouritePokemon: 'Pikachu',
};

export {Favourites};
