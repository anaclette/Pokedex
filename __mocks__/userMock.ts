import {Auth} from '../src/state/reducers/authReducer';

const User: Auth = {
  username: 'Daniel',
};

export {User};
