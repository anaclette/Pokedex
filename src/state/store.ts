import {combineReducers, configureStore} from '@reduxjs/toolkit';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {persistReducer, persistStore} from 'redux-persist';
import authReducer from './reducers/authReducer';
import pokemonsReducer from './reducers/pokemonsReducer';
import favouritesReducer from './reducers/favouritesReducer';
import {STATE_MODULES} from '../common/constants';
import {i18nReducer} from './reducers/i18nReducer';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};

const rootReducer = combineReducers({
  [STATE_MODULES.AUTH]: authReducer,
  [STATE_MODULES.POKEMONS]: pokemonsReducer,
  [STATE_MODULES.FAVOURITES]: favouritesReducer,
  [STATE_MODULES.LANGUAGES]: i18nReducer.reducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      immutableCheck: false,
      serializableCheck: false,
    }),
});

const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export {persistor, store};
