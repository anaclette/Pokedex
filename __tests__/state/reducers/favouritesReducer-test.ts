import authReducer, {
  toggleIsFavourite,
  cleanUpFavourites,
} from '../../../src/state/reducers/favouritesReducer';
import {store} from '../../../src/state/store';
import {mockNewListPokemon} from '../../../__mocks__/pokemonMocks';
import {Favourites} from '../../../__mocks__/favouritesMock';

describe('Testing favourites reducer', () => {
  it('should show stored favourites', async () => {
    await store.dispatch(toggleIsFavourite(mockNewListPokemon));
  });
});
describe('Testing having removed all favourites', () => {
  it('should be none favourites stored', () => {
    const initialState = {
      data: [],
      favouritePokemon: '',
    };
    expect(authReducer(Favourites, {type: cleanUpFavourites})).toEqual(
      initialState,
    );
  });
});
