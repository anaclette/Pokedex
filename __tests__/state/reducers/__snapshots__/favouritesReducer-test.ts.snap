// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing having removed all favourites should be none favourites stored 1`] = `
Object {
  "data": Array [],
  "favouritePokemon": "",
}
`;
